### PR TITLE
Add option to provide KDTree's 'mask' argument when querying

### PR DESCRIPTION
**Pull request by djhoese**
Also remove scipy kdtree as an option and removed get_capabilities. Needs tests and documentation. Also will need pykdtree 1.3.1 when it is released.

 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/master **/*py | flake8 --diff`` 
 - [x] Fully documented

### CODE REVIEW

**Comment by djhoese** [comment]
I also plan to follow this up with a completely dask-ified nearest neighbor resampling that shouldn't have to compute anything except when caching...I hope.

File: pyresample/kd_tree.py [hunk 1]
```diff
@@ -510,12 +492,10 @@ def _create_resample_kdtree(source_lons,
         raise EmptyResult('No valid data points in input data')
 
     # Build kd-tree on input
-    if kd_tree_name == 'pykdtree':
-        resample_kdtree = KDTree(input_coords)
-    elif nprocs > 1:
+    if nprocs > 1:
```
Comment (mraspaud) [inline pyresample/kd_tree.py]: I think this case can be removed too

File: pyresample/kd_tree.py [hunk 2]
```diff
@@ -1061,7 +1040,9 @@ def get_neighbour_info(self):
                             (target_lats <= 90) & (target_lats >= -90))
 
         index_arr, distance_arr = self._query_resample_kdtree(
-            resample_kdtree, target_lons, target_lats, valid_output_idx)
+            resample_kdtree, target_lons, target_lats, valid_output_idx,
+            # None)
+            mask.data.ravel()[valid_input_idx.ravel()])
```
Comment (mraspaud) [inline pyresample/kd_tree.py]: in satpy, `get_neighbour_info` is called only once. So this query call should probably be moved to `get_pixels_from_neighbour_info`

File: pyresample/kd_tree.py (hunk 1 above)
Comment (djhoese) [reply pyresample/kd_tree.py]: We should probably remove the `nprocs` option all together then. Right?

Comment (mraspaud) [reply pyresample/kd_tree.py]: yep

File: pyresample/kd_tree.py (hunk 2 above)
Comment (djhoese) [reply pyresample/kd_tree.py]: You mean `get_sample_from_neighbour_info`? But then that changes the interface and changes what gets cached (if `mask` is None).

File: pyresample/geometry.py [hunk 3]
```diff
@@ -195,6 +195,20 @@ def get_lonlats(self, data_slice=None, **kwargs):
         else:
             return self.lons[data_slice], self.lats[data_slice]
 
+    def get_lonlats_dask(self, chunks=CHUNK_SIZE):
```
Comment (mraspaud) [inline pyresample/geometry.py]: Why move this function ?

File: pyresample/kd_tree.py [hunk 4]
```diff
@@ -1072,54 +1059,149 @@ def get_neighbour_info(self):
                 self.distance_array)
 
     def get_sample_from_neighbour_info(self, data, fill_value=np.nan):
+        """Get the pixels matching the target area.
```
Comment (mraspaud) [inline pyresample/kd_tree.py]: This function got a bit long...

File: pyresample/geometry.py (hunk 3 above)
Comment (djhoese) [reply pyresample/geometry.py]: It is used for SwathDefinitions, AreaDefinitions, and GridDefinitions. Previously it was not defined for GridDefinitions which are perfectly acceptable input area definitions.

**Pull request merged**